pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
What about DELETE of the collection?  And LIST?  And WATCH?  That's 3 new special methods.

----- Original Message -----
> >  		ctx := scope.ContextFunc(req)
> >  		ctx = api.WithNamespace(ctx, namespace)
> > +		if len(name) > 0 {
> > +			ctx = api.WithValue(ctx, "name", name)
> 
> I think we're saying the same thing. I was thinking the new create method
> would take a runtime.Object and a name.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7718/files#r29632731